src/com/civpvp/attrhider/AttrHider.java

```diff
@@ -33,129 +26,84 @@
-	private ProtocolManager protocolManager;
-
+public class AttrHider extends JavaPlugin {
+	
 	@Override
 	public void onEnable() {
-	    registerPacketListeners();
-	    Bukkit.getPluginManager().registerEvents(this, this);
+		registerPacketAdapters();
 	}
 	
-	private void registerPacketListeners(){
-		protocolManager = ProtocolLibrary.getProtocolManager();
-
-        //Strips armour
-        protocolManager.addPacketListener(new PacketAdapter(this, PacketType.Play.Server.ENTITY_EQUIPMENT) {
-            @Override
-            public void onPacketSending(PacketEvent e) {
-                try {
-                    PacketContainer p = e.getPacket();
-                    StructureModifier<ItemStack> items = p.getItemModifier();
-                    ItemStack i = items.read(0);
-                    if (i != null && shouldBeObfuscated(i.getType())) {
-                    	Color color = null;
-                    	if (i.getItemMeta() instanceof LeatherArmorMeta) {
-                    		LeatherArmorMeta lam = (LeatherArmorMeta) i.getItemMeta();
-                    		color = lam.getColor();
-                    	}
-                    	ItemStack is = new ItemStack(i.getType(), 1 , (short) 1);
-                    	if (i.getItemMeta().getEnchants().values().size() != 0) {
-                    		is.addEnchantment(Enchantment.DURABILITY, 1);
-                    	}
-                    	if (color != null) {
-                    		LeatherArmorMeta lam = (LeatherArmorMeta) is.getItemMeta();
-                    		lam.setColor(color);
-                    		is.setItemMeta(lam);
-                    	}
-                        items.write(0, is);
-                    }
-                } catch (FieldAccessException exception) { //Should catch if the packet is the wrong type
-                    exception.printStackTrace();
-                }
-            }
-        });
-
-        //Strips potion duration length and sets it to 420 ticks so you can blaze it
-	    protocolManager.addPacketListener(new PacketAdapter(this, PacketType.Play.Server.ENTITY_EFFECT) {
-	    	@Override
-	    	public void onPacketSending(PacketEvent e) {
-	    		try {
-		    		PacketContainer p = e.getPacket();
-		    		if(e.getPlayer().getEntityId() != p.getIntegers().read(0)) { //Don't hide a player's own stuff from them
-		    			p.getIntegers().write(1, 420);
-		    		}
-	    		} catch (FieldAccessException exception) { 
-	    			exception.printStackTrace();
-	    		}
-	    	}
-	    });
-	    
-	    //Make reported health random
+	public void registerPacketAdapters() {
+		ProtocolLibrary.getProtocolManager().addPacketListener(
+			new PacketAdapter(this, ListenerPriority.NORMAL, PacketType.Play.Server.ENTITY_EQUIPMENT) {
+				@Override
+				public void onPacketSending(PacketEvent event) {
+					ItemStack old = event.getPacket().getItemModifier().read(0);
+					if(old == null) return;
+					ItemStack clean = new ItemStack(old.getType());
```

Comment:
You forgot durability here, this will always reset durability of items sent to 0, which is bad for any block or item that has variations. Tools and armour should have their dura set to a fixed value though. That is the reason why the shouldBeObfuscated method exists and is still needed.